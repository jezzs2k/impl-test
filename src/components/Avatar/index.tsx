import React from 'react';
import {Image, TouchableOpacity, ViewStyle, StyleProp} from 'react-native';

import {widthPercentageToDP} from '../../utils';

export const LINK_IMAGE_DEFAULT =
  'https://interactive-examples.mdn.mozilla.net/media/cc0-images/grapefruit-slice-332-332.jpg';

type AvatarPropType = {
  uri?: string;
  size?: number;
  styleAvatar?: StyleProp<ViewStyle>;
  borderWidth?: number;
  onPress?: () => void;
  activeOpacity?: number;
  borderColor?: string;
};

export const Avatar = ({
  uri,
  size,
  styleAvatar,
  borderWidth,
  onPress,
  activeOpacity,
  borderColor,
}: AvatarPropType) => (
  <TouchableOpacity
    style={[styleAvatar]}
    onPress={onPress}
    accessibilityRole={'button'}
    activeOpacity={activeOpacity}>
    <Image
      source={{uri}}
      style={{
        width: size,
        height: size,
        borderRadius: size / 2,
        borderWidth: borderWidth,
        borderColor: borderColor,
      }}
    />
  </TouchableOpacity>
);

Avatar.defaultProps = {
  uri: LINK_IMAGE_DEFAULT,
  size: widthPercentageToDP(20),
  styleAvatar: {},
  borderWidth: 0,
  onPress: () => {},
  activeOpacity: 1,
  borderColor: '#fff',
};
