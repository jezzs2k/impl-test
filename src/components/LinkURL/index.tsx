import React, {useCallback} from 'react';
import {
  Linking,
  StyleProp,
  Text,
  TouchableOpacity,
  TextStyle,
  StyleSheet,
} from 'react-native';
import {responsiveFontSetting} from '../../utils';

type LinkURLProptype = {
  url: string;
  styleUrl?: StyleProp<TextStyle>;
  title?: string;
};

export const LinkURL = ({
  url,
  styleUrl = {},
  title = 'Brower',
}: LinkURLProptype) => {
  const handlePress = useCallback(async () => {
    const supported = await Linking.canOpenURL(url);

    if (supported) {
      await Linking.openURL(url);
    }
  }, [url]);

  return (
    <TouchableOpacity onPress={handlePress}>
      <Text style={[styles.textSmall, styles.textBold, styles.link, styleUrl]}>
        {title}
      </Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  textBold: {
    fontWeight: '700',
    fontSize: responsiveFontSetting(2.5),
  },
  link: {
    fontStyle: 'italic',
    textTransform: 'capitalize',
    fontSize: responsiveFontSetting(2),
    color: '#1068bf',
    textDecorationLine: 'underline',
  },
  textSmall: {
    fontSize: responsiveFontSetting(1.6),
    fontStyle: 'normal',
  },
});
