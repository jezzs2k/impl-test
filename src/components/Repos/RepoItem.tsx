import React from 'react';
import {View, Text, TouchableOpacity, StyleSheet} from 'react-native';
import AntDesign from 'react-native-vector-icons/AntDesign';
import {StackActions, useNavigation} from '@react-navigation/native';

import {
  heightPercentageToDP,
  responsiveFontSetting,
  widthPercentageToDP,
} from '../../utils';
import {ShowTextItem} from '../Text/ShowTextItem';

type RepoItemProptype = {
  item: any;
};

export const RepoItem = ({item}: RepoItemProptype) => {
  const navigation = useNavigation();

  const {
    name,
    open_issues,
    forks,
    watchers,
    description,
    stargazers_count,
    stargazers_url,
  } = item || {
    open_issues: 0,
    forks: 0,
    watchers: 0,
    description: null,
    stargazers_count: 0,
  };

  const pushAction = StackActions.push('Stargazers', {
    urlStargazer: stargazers_url,
  });

  const handleToStargazers = () => {
    navigation.dispatch(pushAction);
  };

  return (
    <View style={styles.container}>
      <View style={styles.wrapperText}>
        <ShowTextItem
          keyItem="Repositories"
          title={name}
          textStyle={styles.textNomal}
          subTextStyle={styles.subTextStyle}
        />
        <ShowTextItem
          keyItem="Descriptions"
          title={description}
          textStyle={styles.textNomal}
          subTextStyle={styles.subTextStyle}
        />
        <ShowTextItem
          keyItem="Forks"
          title={forks}
          textStyle={styles.textNomal}
          subTextStyle={styles.subTextStyle}
        />
        <ShowTextItem
          keyItem="Watchers"
          title={watchers}
          textStyle={styles.textNomal}
          subTextStyle={styles.subTextStyle}
        />
        <ShowTextItem
          keyItem="Open Issues"
          title={open_issues}
          textStyle={styles.textNomal}
          subTextStyle={styles.subTextStyle}
        />
      </View>
      <TouchableOpacity
        style={styles.containerItem}
        onPress={handleToStargazers}
        disabled={stargazers_count <= 0}>
        <Text style={styles.textNomal}>
          Stargazers: <Text style={styles.textBold}>{stargazers_count}</Text>
        </Text>
        <AntDesign
          name={'right'}
          size={responsiveFontSetting(2)}
          color={'#aaa69d'}
        />
      </TouchableOpacity>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    marginTop: heightPercentageToDP(1),
    backgroundColor: '#fff',
    paddingVertical: heightPercentageToDP(0.5),
    flexDirection: 'row',
    justifyContent: 'space-between',
    paddingHorizontal: widthPercentageToDP(5),
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 3.84,
    elevation: 5,
  },
  wrapperText: {
    maxWidth: widthPercentageToDP(55),
  },
  subTextStyle: {
    color: '#2c3e50',
    fontSize: responsiveFontSetting(1.8),
  },
  textStyle: {
    fontSize: responsiveFontSetting(2),
  },
  textBold: {
    fontWeight: '700',
    fontSize: responsiveFontSetting(1.8),
  },
  textNomal: {
    fontSize: responsiveFontSetting(2),
    fontWeight: '200',
    fontStyle: 'italic',
  },
  containerItem: {
    flexDirection: 'row',
    alignItems: 'center',
  },
});
