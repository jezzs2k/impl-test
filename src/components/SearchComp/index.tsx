import React from 'react';
import {View, TextInput, Text, StyleSheet} from 'react-native';
import {
  heightPercentageToDP,
  responsiveFontSetting,
  widthPercentageToDP,
} from '../../utils';

type SearchCompPropType = {
  onchangeText: (text: string) => void;
  textTitle: string;
  placeholder?: string;
};

export const SearchComp = ({
  onchangeText,
  textTitle,
  placeholder,
}: SearchCompPropType) => {
  return (
    <View style={styles.container}>
      <View style={styles.containerItem}>
        <Text style={styles.textBold}>{textTitle}</Text>
      </View>
      <TextInput
        style={styles.textInputStyle}
        placeholder={placeholder ?? 'Search user ...'}
        onChangeText={onchangeText}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#fff',
    paddingHorizontal: widthPercentageToDP(5),
    paddingVertical: heightPercentageToDP(1),
  },
  containerItem: {
    marginBottom: heightPercentageToDP(1),
    paddingTop: heightPercentageToDP(1),
  },
  textStyle: {
    fontSize: responsiveFontSetting(2),
  },
  textBold: {
    fontWeight: '700',
    fontSize: responsiveFontSetting(2.5),
    textTransform: 'capitalize',
  },
  textInputStyle: {
    paddingVertical: heightPercentageToDP(1),
    paddingHorizontal: widthPercentageToDP(3),
    borderWidth: 1,
    fontSize: responsiveFontSetting(2),
    borderRadius: 8,
    borderColor: '#bdc3c7',
  },
});
