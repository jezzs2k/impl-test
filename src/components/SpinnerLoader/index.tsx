import React from 'react';
import {View, StyleSheet} from 'react-native';
import Svg, {Rect} from 'react-native-svg';
import ContentLoader from 'react-native-masked-loader';

const getMaskedElement = () => {
  return (
    <Svg height={'250px'} width="100%" fill={'black'}>
      <Rect x="0" y="0" rx="8" ry="8" width="50%" height="16" />
      <Rect x="0" y="30" rx="9" ry="9" width="100%" height="128" />
      <Rect x="0" y="172" rx="4" ry="4" width="100%" height="8" />
      <Rect x="0" y="188" rx="4" ry="4" width="100%" height="8" />
      <Rect x="0" y="204" rx="4" ry="4" width="100%" height="8" />
    </Svg>
  );
};

export const BOSLoader = () => {
  const MaskedElement = getMaskedElement();
  return (
    <View style={styles.container}>
      <ContentLoader MaskedElement={MaskedElement} />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#fff',
    height: 285,
  },
});
