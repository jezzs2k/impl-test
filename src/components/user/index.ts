export * from './userDetailHeader';
export * from './userItem';
