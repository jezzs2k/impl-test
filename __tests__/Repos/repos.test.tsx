import React from 'react';
import renrender from 'react-test-renderer';
import {render, fireEvent} from '@testing-library/react-native';

import {RepoItem} from '../../src/components/Repos';

describe('Repo Item', () => {
  it('Snapshot test repo item component', () => {
    const component = renrender.create(<RepoItem />);

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
