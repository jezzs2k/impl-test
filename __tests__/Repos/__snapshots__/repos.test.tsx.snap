// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repo Item Snapshot test repo item component 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "elevation": 5,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginTop": 13.5,
      "paddingHorizontal": 37.5,
      "paddingVertical": 6.5,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.25,
      "shadowRadius": 3.84,
    }
  }
>
  <View
    style={
      Object {
        "maxWidth": 412.5,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 13.25,
          },
          Object {
            "fontSize": 15.25,
            "fontStyle": "italic",
            "fontWeight": "200",
          },
          Object {
            "fontSize": 13.25,
            "fontStyle": "italic",
            "fontWeight": "200",
          },
        ]
      }
    >
      Forks
      : 
      <Text
        style={
          Array [
            Object {
              "color": "#2c3e50",
              "fontSize": 14,
              "fontWeight": "700",
            },
            Object {
              "color": "#2c3e50",
              "fontSize": 12,
            },
          ]
        }
      >
        0
      </Text>
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontSize": 13.25,
          },
          Object {
            "fontSize": 15.25,
            "fontStyle": "italic",
            "fontWeight": "200",
          },
          Object {
            "fontSize": 13.25,
            "fontStyle": "italic",
            "fontWeight": "200",
          },
        ]
      }
    >
      Watchers
      : 
      <Text
        style={
          Array [
            Object {
              "color": "#2c3e50",
              "fontSize": 14,
              "fontWeight": "700",
            },
            Object {
              "color": "#2c3e50",
              "fontSize": 12,
            },
          ]
        }
      >
        0
      </Text>
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontSize": 13.25,
          },
          Object {
            "fontSize": 15.25,
            "fontStyle": "italic",
            "fontWeight": "200",
          },
          Object {
            "fontSize": 13.25,
            "fontStyle": "italic",
            "fontWeight": "200",
          },
        ]
      }
    >
      Open Issues
      : 
      <Text
        style={
          Array [
            Object {
              "color": "#2c3e50",
              "fontSize": 14,
              "fontWeight": "700",
            },
            Object {
              "color": "#2c3e50",
              "fontSize": 12,
            },
          ]
        }
      >
        0
      </Text>
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 13.25,
          "fontStyle": "italic",
          "fontWeight": "200",
        }
      }
    >
      Stargazers: 
      <Text
        style={
          Object {
            "fontSize": 12,
            "fontWeight": "700",
          }
        }
      >
        0
      </Text>
    </Text>
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#aaa69d",
            "fontSize": 13.25,
          },
          undefined,
          Object {
            "fontFamily": "anticon",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      î˜Ÿ
    </Text>
  </View>
</View>
`;
